infra_event_notifier/backends/jira.py

```diff
@@ -1,8 +1,120 @@
-from typing import Mapping
+from typing import cast, Optional, Sequence
+import requests
+from requests.auth import HTTPBasicAuth
+
+
+MAX_JIRA_DESCRIPTION_LENGTH = 32000
+
+class JiraConfig:
+    def __init__(
+        self, url: str, project_key: str, user_email: str, api_token: str
+    ):
+        self.url = url
+        self.project_key = project_key
+        self.user_email = user_email
+        self.api_token = api_token
+
+
+class JiraApiException(Exception):
+    def __init__(self, stderr: str) -> None:
+        self.message = stderr
+        super().__init__(self.message)
 
 
 def create_issue(
-    title: str, text: str, tags: Mapping[str, str], jira_api_key: str
-) -> None:
-    # TODO: implement
-    pass
+    title: str, text: str, labels: Sequence[str], jira: JiraConfig
+) -> requests.Response:
+    api_url = f"{jira.url}/rest/api/2/issue"
+
+    issue_data = {
+        "fields": {
+            "project": {"key": jira.project_key},
+            "summary": title,
+            "description": text[:MAX_JIRA_DESCRIPTION_LENGTH],
```

Comment:
we can either truncate the message here or just fail fast if text exceeds length